Xamarin.Forms.Platform.Android/Renderers/EditorRenderer.cs

```diff
@@ -56,12 +56,28 @@ protected override FormsEditText CreateNativeControl()
 			return new FormsEditText(Context);
 		}
 
+		protected override void OnFocusChangeRequested(object sender, VisualElement.FocusRequestArgs e)
+		{
+			if (!e.Focus)
+			{
+				Control.HideKeyboard();
+			}
+
+			base.OnFocusChangeRequested(sender, e);
+
+			if (e.Focus)
+			{
+				// Post this to the main looper queue so it doesn't happen until the other focus stuff has resolved
+				// Otherwise, ShowKeyboard will be called before this control is truly focused, and we will potentially
+				// be displaying the wrong keyboard
+				Device.BeginInvokeOnMainThread(() => Control.ShowKeyboard());
```

Comment:
The code in `OnFocusChangeRequested` seems to be repeated across renderers. Should it be refactored into a separate file/method?

Reply:
We'd have to either create two shared methods for the code on either side of `base.OnFocusChangeRequested(sender, e);` or pass the call to `base` in as a delegate. I don't love the first option because it creates two very small methods that don't do much and doesn't reduce the actual code duplication by much. And the second option introduces a new and possibly confusing pattern here. 

If we were talking about more code (say, 50 lines instead of 11 (and mostly braces/comments)), I'd say yes. But I'm not sure the maintenance gain is worth the complexity penalty here.